### ISSUE TITLE
Changes to editor_help.cpp make the built in editor help unreadable  on smaller or scaled screens

### ISSUE THREAD

**Issue by ghost** [godotengine/godot#55875]
### Godot version

3.x (1cf6d21)

### System information

Windows 10 GLES3/2

### Issue description

The changes made in https://github.com/godotengine/godot/pull/55560 to `editor_help.cpp` make the page unreadable on smaller or scaled screens.  

There is no need to have 3 columns in the editor help because some of the `[default]` options can be very long resource path names pushing the other columns and making them unreadable. 

The original solution worked better. 

ProjectSettings.xml on 1920 x 1080 and 125% scaling with this pr
![image](https://user-images.githubusercontent.com/36826319/145729306-fc43537f-f9b8-4fc4-9672-60159dcafa9c.png)

### Steps to reproduce

Open Project Settings with the latest 3.x using scaled screen

**Comment by Calinou** [comment]
cc @pycbouh

**Comment by pycbouh** [comment]
> There is no need to have 3 columns in the editor help because some of the `[default]` options can be very long resource path names pushing the other columns and making them unreadable.
> 
> The original solution worked better.

Overall, I don't agree with this. It didn't work better for the majority of classes. But it's hard to find a good solution that covers all edge cases. Like the ProjectSettings docs which have abnormally long property names.

I'm not sure how text rendering for tables works in 3.x, but it shouldn't really break types onto new lines like that, it has plenty of spaces for them. And the gaps can be reduced between columns to prevent to depicted property name from line wrapping. But I'm not confident that we can control these aspects of text rendering in the text label as granularly.

**Comment by filipworksdev** [comment]
Anytime you have a long property or a `[default: "res://with_a_long_resource_name.tres"]` will crowd all other columns like you see above. 

In the screen shot above this
```
    application/config/use_hidden_project_data_directo
ry
```
get croded because of this  further down
```
    [default: 
"res://default_bus_layout.tres"]
```


I see 3 main issues with your solution:
1. How will you detect edge cases? Is there some algorithm that can detect visual crowding? Before crowding was avoided because because it would sort itself out having more space in column 2 to put the long property name on one line and the default on the line beneath but with 3 columns this cannot be avoided. This issue can pop up anywhere in modules or addons.
2. Docs will no longer have a consistent look as edge case pages now look differently.
3. You will have two different code for two different ways of rendering the same thing.  I thought the idea was to have one thing work one way only.

**Comment by pycbouh** [comment]
> You will have two different code for two different ways of rendering the same thing. I thought the idea was to have one thing work one way only.

If this was true, I would rather we have a solution that helps make the majority of the docs more readable rather than a solution that handicaps the rest for the benefit of a single outlier. And this is the case of a single outlier, as only ProjectSettings realistically can have such long-long names because that's essentially setting paths stored as properties.

Thankfully, that's not true. It makes sense to add a special case for ProjectSettings because of its unique nature and make it work better for its unique composition of long name and also, sometimes, loner values.

**Comment by filipworksdev** [comment]
Hello. 

So I did more testing.  OS  singleton shows crowding as well for me.  AnimationPlayer too a bit if I have both Scripts tab and the other tabs extended too much. Probably other pages as well. Editor minimized crowding is much worse now.

AABB on editor not maximized with pr (default breaks off in 3 rows)
![image](https://user-images.githubusercontent.com/36826319/145737524-4d8d22eb-7fb2-420d-ac76-64ce18f40fcb.png)

and without pr (default break off in 2 rows)
![image](https://user-images.githubusercontent.com/36826319/145740357-5e3c9770-c5df-4d7e-9cd0-b2a2d8404090.png)

To me the bottom one is easier to read since there's less line breaking. I can read continously for longer.  Lines should break off at words but code should break off logically otherwise is hard to read.  With 3 colums there's just not enough space for long lines to grow into.

A better approach would be to keep the 2 columns and have the default move to the 2nd line if there's no more room on the first. This would give more room for long properties to grow into and potentially line breaking. 

Broken lines are super hard to read. I assume people with disabilities would struggle even more.  I personally don't care if it was 2 columns or 3 but reading code like this is really difficult
```
this_is(som
e, code wh
ich) is [ver
y, long]
``` 

Also the idea to hardcode doc names and  implement alternative layouting just for a couple of edge cases I dunno why but I would not take that approach. Doesn't seem future proof.  I think a better approach would be revert and explore a more solid alternative that is equivalent to or better than the previous implementation.  Anyway that's all I had to say.  Hope you guys acn figure it out.

**Comment by pycbouh** [comment]
The editor UI will break if you keep making the window smaller and smaller, so I'm not sure what you are trying to demonstrate with those last examples...

**Comment by filipworksdev** [comment]
Hi thanks for the reply. 

Sorry I guess I should have explained myself more correctly. So just to re-iterate here are some of the points I was trying to demonstrate:
- Changes to editor_help.cpp makes the doc pages break more easily and overall broken lines look worse than before.
- In order to get docs to render correctly you now need at least a full HD screen, no scaling and Godot Editor to be maximized. 
- ProjectSettings is biggest offender but I can see it also in OS and AnimationPlayer and pretty much anywhere when editor is not maximized.
- When editor is not maximized docs looks more crowded and harder to read than before. 
- Your solution to hardcode some pages and render them differently doesn't sound like a permanent solution and cannot account for docs from addons/modules. 

Hope that is more clear. Thanks 😊

**Comment by akien-mga** [comment]
I don't mind much what kind of solution is used, but I agree that there are cases where the docs break more easily now than they used to, and that's particularly visible whenever there are very long types, or very long property names, or very long default values.

It would be good to find an approach that could reduce the frequency of these issues for 3.5, and maybe also tweak the format so that when wrapping does happen, it does not look weird like now.

ProjectSettings in 3.4.4-stable in 1080p (no scaling):

![image](https://user-images.githubusercontent.com/4701338/166825940-20305b71-dfc6-4b83-ba71-08c899d33716.png)

ProjectSettings in current `3.x` (37fd526dddc610ff6b48fd2084c1c6a678c47b54):

![image](https://user-images.githubusercontent.com/4701338/166826101-a27cb791-02a4-4b54-9ca2-e56f75a331d7.png)

The broken "Dictiona" / "ry" is really bad. "PoolStri" / "ngArray" is also annoying, and it breaks in a different pattern for some reason.

![image](https://user-images.githubusercontent.com/4701338/166826307-a220f7c4-3a23-411b-b1ba-ac73fbacdcf1.png)

This is also weird but not horrible, it's just weird that the wrapped lines are dedented.

![image](https://user-images.githubusercontent.com/4701338/166826458-13aacbc1-8f3b-4f08-8e64-6544b18776f8.png)

**Comment by akien-mga** [comment]
Fixed by #61904.
